libr/core/cconfig.c

```diff
@@ -33,6 +33,37 @@ static void set_options(RConfigNode *node, ...) {
 	va_end (argp);
 }
 
+static bool set_arch(RCore *c, const char *name) {
+	RArch *a = c->arch;
+	RArchPlugin *ap = r_arch_get_plugin (a, name);
+	if (ap) {
+		RArchSetup setup = {
+			.bits = r_config_get_i (c->config, "asm.bits"),
+			.cpu = strdup (r_config_get (c->config, "asm.cpu")),
+			.endian = r_config_get_i (c->config, "cfg.bigendian")? R_SYS_ENDIAN_BIG: R_SYS_ENDIAN_LITTLE,
+			.syntax = r_config_get_i (c->config, "asm.syntax"),
+		};
+		RArchSession *as = r_arch_session_new (a, ap, &setup);
+		if (as) {
+			r_arch_session_unref (c->assembler->asd);
+			c->assembler->asd = as;
```

Comment:
it took me a while to understand what `asd` meant. I thought it was random name :P